File: src/swift/queue.swift
//
//  fila.swift
//
//  Created by Matheus Torres on 14/11/20.
//

class Queue {
    var queue: [Int] = []
    
    func enqueue(_ value: Int) {
        self.queue.append(value)
    }
    
    func dequeue() {
        self.queue.removeFirst()
    }
    
    func show() {
        print(self.queue)
    }
    
    func clear() {
        self.queue = []
    }
}

let queue = Queue()

for n in 0...10 { queue.enqueue(n) }
queue.show()
queue.dequeue()
queue.show()

File: src/javascript/Queue.js
class Queue {
    constructor() {
        this.queue = [];
    }

    enqueue(value) {
        this.queue.push(value);
    }

    dequeue() {
        this.queue.shift(0);
    }

    read() {
        console.log(this.queue);
    }

    empty() {
        this.queue = [];
    }
}

function main() {
  let queue = new Queue();

  queue.enqueue(1);
  queue.enqueue(2);
  queue.enqueue(3);
  queue.read();

  queue.dequeue();
  queue.read();

  queue.empty();
  queue.read();
}

main();
